op-supervisor/supervisor/backend/db/fromda/entry.go

```diff
@@ -0,0 +1,96 @@
+package fromda
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/db/entrydb"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+const EntrySize = 100
+
+type Entry [EntrySize]byte
+
+func (e Entry) Type() EntryType {
+	return EntryType(e[0])
+}
+
+type EntryType uint8
+
+const (
+	DerivedFromV0 EntryType = 0
+)
+
+func (s EntryType) String() string {
+	switch s {
+	case DerivedFromV0:
+		return "v0"
+	default:
+		return fmt.Sprintf("unknown(%d)", uint8(s))
+	}
+}
+
+type EntryBinary struct{}
+
+func (EntryBinary) Append(dest []byte, e *Entry) []byte {
+	return append(dest, e[:]...)
+}
+
+func (EntryBinary) ReadAt(dest *Entry, r io.ReaderAt, at int64) (n int, err error) {
+	return r.ReadAt(dest[:], at)
+}
+
+func (EntryBinary) EntrySize() int {
+	return EntrySize
+}
+
+type LinkEntry struct {
+	derivedFrom types.BlockSeal
+	derived     types.BlockSeal
+}
```

Comment:
nit: would prefer a slightly more verbose name that includes the Derivation specific application. `DerivationLinkEntry`?